src/Interpreters/InterpreterShowIndexesQuery.cpp

```diff
@@ -24,65 +23,62 @@ String InterpreterShowIndexesQuery::getRewrittenQuery()
-                            << "NULL AS seq_in_index, "
-                            << "NULL AS column_name, "
-                            << "'A' AS collation, "
-                            << "NULL AS cardinality, "
-                            << "NULL AS sub_part, "
-                            << "NULL AS packed, "
-                            << "NULL AS null, "
-                            << "'primary' AS index_type, "
-                            << "NULL AS comment, "
-                            << "NULL AS index_comment, "
-                            << "'YES' AS visible, "
-                            << "primary_key AS expression "
-                        << "FROM system.tables "
-                        << "WHERE "
-                            << "database = '" << database << "' "
-                            << "AND name = '" << table << "'"
-                    << ") UNION ALL ("
-                        << "SELECT "
-                            << "table AS table, "
-                            << "0 AS non_unique, "
-                            << "name AS key_name, "
-                            << "NULL AS seq_in_index, "
-                            << "NULL AS column_name, "
-                            << "NULL AS collation, "
-                            << "NULL AS cardinality, "
-                            << "NULL AS sub_part, "
-                            << "NULL AS packed, "
-                            << "NULL AS null, "
-                            << "type AS index_type, "
-                            << "NULL AS comment, "
-                            << "NULL AS index_comment, "
-                            << "'YES' AS visible, "
-                            << "expr AS expression "
-                        << "FROM system.data_skipping_indices "
-                        << "WHERE "
-                            << "database = '" << database << "' "
-                            << "AND table = '" << table << "'"
-                            << ")) "
-                        << where_expression;
+    String table = query.table;
+    String database = query.database.empty() ? getContext()->getCurrentDatabase() : query.database;
+    String where_expression = query.where_expression ? fmt::format("WHERE ({})", query.where_expression) : "";
+
+    String rewritten_query = fmt::format(R"(
+SELECT *
+FROM (
+        (SELECT
+            name AS table,
+            0 AS non_unique,
+            'PRIMARY' AS key_name,
+            NULL AS seq_in_index,
+            NULL AS column_name,
+            'A' AS collation,
+            NULL AS cardinality,
+            NULL AS sub_part,
+            NULL AS packed,
+            NULL AS null,
+            'primary' AS index_type,
+            NULL AS comment,
+            NULL AS index_comment,
+            'YES' AS visible,
+            primary_key AS expression
+        FROM system.tables
+        WHERE
+            database = '{0}'
```

Comment:
SQL injection.

Reply:
Add a test with the database name and table name containing the `'` character.

Reply:
The parsers of `SHOW INDEX` and `SHOW COLUMNS` expect database and table as identifiers, whereas `WHERE table/database =` expects a string literal. The mismatch is now properly resolved using quoting. I have also added a test.